irc/channel.go

```diff
@@ -407,7 +407,9 @@ func (channel *Channel) Join(client *Client, key string, rb *ResponseBuffer) {
 	// give channel mode if necessary
 	newChannel := firstJoin && !channel.IsRegistered()
 	var givenMode *modes.Mode
-	if client.AccountName() == channel.registeredFounder {
+	account := client.Account()
+	cffounder, _ := CasefoldName(channel.registeredFounder)
```

Comment:
Here, are we able to remove the casefolding on channel.registeredFounder? (if so all good, just wanna be sure)

Reply:
There's a backwards compatibility issue here. As of this branch, the account name is stored casefolded in the database (and in the in-memory `registeredFounder` field). However, previously it was stored unfolded:

https://github.com/oragono/oragono/blob/fcd0a754695f3699072b80e0fece218c84a23130/irc/chanserv.go#L79

In theory, removing the casefolding here would break backwards compatibility. I can see three solutions:

1. Leave as-is, with a comment
1. Go through the schema change process to enforce that `registeredFounder` is always unfolded
1. Remove the casefolding here and break backwards compatibility (it doesn't affect my deployment, that's for sure)

Reply:
Ah, heh. That's fun.

For what it's worth, I prefer having everything folded, and comparing folded names always where possible.

If previously, this name was stored in the database unfolded, we should go through a schema update that forces all the names to be folded, yeah.